src/database/sql_database.h

```diff
@@ -270,11 +270,11 @@ class SQLDatabase : public Database {
     /* helper for removeObject(s) */
     void _removeObjects(const std::vector<std::int32_t>& objectIDs);
 
-    std::unique_ptr<ChangedContainers> _recursiveRemove(
+    ChangedContainers _recursiveRemove(
```

Comment:
If not used exactly in the correcty `move`s, it creates extra copies

Reply:
See above. If you try to copy it will fail to compile.